mergify_engine/tests/functional/test_simulator.py

```diff
@@ -0,0 +1,55 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2019 Mehdi Abaakouk <sileht@sileht.net>
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import yaml
+
+from mergify_engine.tests.functional import base
+
+
+class TestEngineV2Scenario(base.FunctionalTestBase):
+    """Mergify engine tests.
+
+    Tests user github resource and are slow, so we must reduce the number
+    of scenario as much as possible for now.
+    """
+
+    def test_simulator(self):
```

Comment:
That definitely needs some more failure scenarios.

Reply:
I added a some negative tests here and in test_config.